xxhr/auth.hpp

```diff
@@ -35,6 +35,17 @@ class Authentication {
     std::string auth_string_;
 };
 
+
+/**
+ * \brief Some Proxy requires a special kind of authentication via NTLM, this allows specifying it.
+ */
+class NTLM : public Authentication {
+  public:
+    NTLM(const std::string& username, const std::string& password)
+            : Authentication{username, password} {}
+};
+
+
```

Comment:
I don't know how that goes into play with the changes regarding CURL but generally speaking I would expect the client to use the kernel functions to respond to a challenge with the current user sessions credentials instead of having the tipi user enter credentials with us. That being said this would work only on windows machines.